TUTORIAL.md

```diff
@@ -0,0 +1,100 @@
+
+Create new Go project and configure a database for it. Make sure that your database is supported [here](README.md#databases)
+For the purpose of this tutorial let's create PostgreSQL database called `example`.
+Our user here is `postgres`, and host is `localhost`.
+```
+psql -h localhost -U postgres -w -c "create database example;"
+```
+When using Migrate CLI we need to pass to database URL. Let's export it to a variable for convienience:
+```
+export POSTGRESQL_URL=postgres://postgres:postgres@localhost:5432/example?sslmode=disable
+```
+`sslmode=disable` means that the connection with out database will not be encrypted. Enabling it is left as an exercise.
+
+You can find further description of database URLs [here](README.md#database-urls).
+## Create migrations
+Let's create table called `users`:
+```
+migrate create -ext sql -dir db/migrations create_users_table
+```
+If there were no errors, we should have two files available under `db/migrations` folder. Note the `sql` extension that we provided.
+In the `.up.sql` file we are going to create table:
+```
+CREATE TABLE users(
+   user_id serial PRIMARY KEY,
+   username VARCHAR (50) UNIQUE NOT NULL,
+   password VARCHAR (50) NOT NULL,
+   email VARCHAR (300) UNIQUE NOT NULL
+);
+```
+And in the `.down.sql` we are going to delete it:
+```
+DROP TABLE users;
+```
+
+## Run migrations
+```
+migrate -database ${POSTGRESQL_URL} -path db/migrations up
+```
+Let's check if the table was created properly by running `psql example -c "\d users"`.
+The output you are supposed to see:
+```
+                                    Table "public.users"
+  Column  |          Type          |                        Modifiers                        
+----------+------------------------+---------------------------------------------------------
+ user_id  | integer                | not null default nextval('users_user_id_seq'::regclass)
+ username | character varying(50)  | not null
+ password | character varying(50)  | not null
+ email    | character varying(300) | not null
+Indexes:
+    "users_pkey" PRIMARY KEY, btree (user_id)
+    "users_email_key" UNIQUE CONSTRAINT, btree (email)
+    "users_username_key" UNIQUE CONSTRAINT, btree (username)
+```
+Great! Now let's check if rollback also works:
+```
+migrate -database ${POSTGRESQL_URL} -path db/migrations down
+```
+Make sure to check if your database changed as expected on rollback.
+
+**Before commiting your migrations:** You should run your migrations, rollback them, and then run them again to see if migrations are working properly both ways.
+(e.g. if you created a table in a migration but rollback did not delete it, you will encounter an error when running the migration again)
+It's also worth checking your migrations in a separate, containerized environment. You can find some tools in the end of this tutorial.
+
+**Hint:** Most probably you are going to use to the above commands often, it's worth putting them in a Makefile of your project.
```

Comment:
Could you provide more background on why you found wrapping the CLI usage in a Makefile useful?
I've been using the CLI directly.

Reply:
In our project at work, we use it to call `make migrate-up POSTGRESQL_DB=example_test steps=2` instead of `migrate -database postgres://user:password@localhost:5432/example_test -path ./migrations/postgres up 2`. It's just slightly shorter compared to putting database URL in `.env` file.

Reply:
Ahhh, I normally have my Go apps use `migrate` as a library and call `migrate.Migrate.Up()` before serving traffic. Managing migrations out-of-app using the CLI makes sense too. e.g. there are many different dev and deploy processes
I wouldn't assume that all projects are using `make` to manage their development and/or deploys.

Reply:
Right, it may be too environment-specific. I will remove this part